models/Item.js

```diff
@@ -2,8 +2,14 @@ import mongoose from "mongoose";
 
 const { Schema } = mongoose;
 
+const categorySchema = new Schema({
+  name: { type: String },
+  icon_src: { type: String },
+});
+
 const itemsSchema = new Schema({
   name: { type: String, required: true },
+  category: { type: categorySchema },
```

Comment:
This schema implies that you define the name and icon for each item individually. If you want to change a category you will run into the problem, that you would need to update each item that is associated with that category. It would be better to save all categories in their own collection and refer to them inside the items via their ObjectIds which they get from MongoDB. Then you can either filter all items via the respective ID when you only want to display the indiviual lists of items separated by category. If you want to display some information about the category with each item instead, you can use a method called join. You then replace the id in your backend when you fetch with the actual data from the category collection every time you fetch for the items. MongoDB can do this for you, have a look into their documentation how to achieve this.